src/charm.py

```diff
@@ -260,7 +263,8 @@ def _on_certificate_invalidated(self, event: CertificateInvalidatedEvent):
     def _on_all_certificates_invalidated(self, event: RelationBrokenEvent) -> None:
         if self.container.can_connect():
             self._stored.certificate = None
-            self._stored.private_key = None
+            secret = self.model.get_secret(label=PRIVATE_KEY_SECRET_LABEL)
+            secret.remove_all_revisions()
```

Comment:
Currently the charm would go into error state if this raises, which I think is good, because Juju would keep auto retrying.
But would this raise if there are no secrets/revisions left?

Reply:
Good point, fixed.